azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/options/AzureTemplateOptions.java

```diff
@@ -76,12 +83,50 @@ public AzureTemplateOptions availabilitySet(String availabilitySetName) {
       return this;
    }
 
+   public AzureTemplateOptions dataDisks(List<Map<String, String>> dataDisksAsMap) {
```

Comment:
The logic to build the DataDisk object from the map is trivial and does not seem to make sense in jclouds. Also, you are limiting it to how you build the object in the options method. It is more flexible to let users build the DataDisk object. In the end, if they know which fields in the map they need, they already know which fields of the DataDisk object to fill, so I don't see the point here.